src/current-user.js

```diff
@@ -506,6 +507,13 @@ export class CurrentUser {
       })
       .then(() => this.userStore.initialize({}))
   }
+
+  disconnect = () => {
+    this.userSubscription.cancel()
+    this.presenceSubscription.cancel()
+    this.cursorSubscription.cancel()
+    forEachObjIndexed(sub => sub.cancel(), this.roomSubscriptions)
```

Comment:
Is there a reason for using `forEachObjIndexed` as opposed to `map`, as it looks like it was in the `teardown` function that's being removed: `map(sub => sub.cancel(), currentUser.roomSubscriptions)` ?

Reply:
`map` produces a resultant array or object depending on what is passed to it, which isn't used..

Reply:
But does it even if you're not assigning anything to its result?

Reply:
yep, I believe it does!